text-generation-inference/server/text_generation_server/utils/logits_process.py

```diff
@@ -59,16 +58,14 @@ def __call__(self, scores):
                 # Compute logprobs
                 self.static_next_logprob.copy_(torch.log_softmax(self.static_warped_scores, -1))
 
-        if scores.shape[-2] == 1:
-            self.static_scores.copy_(scores)
-            self.hpu_graph.replay()
+        self.static_scores.copy_(scores)
+        self.hpu_graph.replay()
 
-            return self.static_warped_scores, self.static_next_logprob
-        else:
-            # 1st token disposal
-            for warper in self.warpers:
-                scores = warper(None, scores)
-            return scores, torch.log_softmax(scores, -1)
+        return self.static_warped_scores, self.static_next_logprob
+        # cpu branch
+        for warper in self.warpers:
+            scores = warper(None, scores)
+        return scores, torch.log_softmax(scores, -1)
```

Comment:
This part of the code will never be reached as we return before, is it expected?

Reply:
yes, the 1st token score also has shape of [1, xxxx], same with the other token. as in https://github.com/huggingface/optimum-habana/pull/408/files#diff-244871853752a7a019853c852803a797a9a14dddb3d9d859ecbae4b316ed6974R658

Reply:
So why keeping these lines then?

Reply:
should I remove it. I see https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/utils/logits_process.py#L66 . Is original code left here for debug?

Reply:
I think so given this code will never be reached (the method always returns before reaching there)